### PR TITLE
[ANCHOR-220] Refactor README

### PR DESCRIPTION
**Pull request by lijamie98**
### Description

- Fix, clean up and improve the `How to Contribute` section 
- Add How to set up development environment
- Add Git guidelines
- Add repository directory structure
- Removed outdated contents

### CODE REVIEW

File: docs/01 - Contributing/A - Development Environment.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+```shell
+127.0.0.1 db
+127.0.0.1 zookeeper
+127.0.0.1 kafka
+127.0.0.1 sep24-reference-ui
+127.0.0.1 reference-server
+127.0.0.1 wallet-server
+127.0.0.1 platform
+127.0.0.1 host.docker.internal
+```
```
Comment (JakeUrban) [inline docs/01 - Contributing/A - Development Environment.md:78]: This has not been my experience. I've gotten everything working without adding these hosts. Can you reproduce?

**Review by JakeUrban — APPROVED** [review]
🙌 This is an awesome improvement. I have one question but other that than this LGTM.

File: docs/01 - Contributing/A - Development Environment.md (hunk 1 above)
Comment (Ifropc) [reply docs/01 - Contributing/A - Development Environment.md:78]: It's for tests

Comment (lijamie98) [reply docs/01 - Contributing/A - Development Environment.md:78]: Yeah, this is for integration and end-2-end tests where the servers communicate with the docker containers.

Comment (lijamie98) [reply docs/01 - Contributing/A - Development Environment.md]: Added the purpose of the DNS entries to the doc. 
https://github.com/stellar/java-stellar-anchor-sdk/pull/1064/files#diff-700ef4c73b87f78486e1b6c614d99548eebc17d61df8cc6613cc1e2b668ab615R71

**Pull request merged**